### PR TITLE
Add `toml` to checking requirements

### LINKED ISSUE #2813: `Checks` CI job fails with missing `toml` dependency
Black in latest release [switched from using `toml` to `tomli`](https://github.com/psf/black/pull/2301). Since `toml` is no longer installed in `Checks` workflow, `blackdoc` will fail the CI job.

## Expected behavior

`toml` should be installed when CI job runs `pip install --progress-bar off -U .[checking]`

## Environment

- Optuna version: 2.9.0.dev0
- Python version: Any
- OS: Any

## Error messages, stack traces, or logs
In `Checks` CI job
```
Run blackdoc . --check --diff
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.11/x64/bin/blackdoc", line 5, in <module>
    from blackdoc.__main__ import main
  File "/opt/hostedtoolcache/Python/3.7.11/x64/lib/python3.7/site-packages/blackdoc/__main__.py", line 12, in <module>
    from .blackcompat import (
  File "/opt/hostedtoolcache/Python/3.7.11/x64/lib/python3.7/site-packages/blackdoc/blackcompat.py", line 9, in <module>
    import toml
ModuleNotFoundError: No module named 'toml'
```
[Example](https://github.com/optuna/optuna/pull/2810/checks?check_run_id=3088486083)

## Steps to reproduce

1.  Push the code to trigger `Checks` CI job

### PR DESCRIPTION
**Pull request by xadrianzetx**
This PR resolves missing dependency in `Checks` CI workflow caused by latest release of Black.

## Motivation
Resolves #2813

## Description of the changes
Added `toml` to `checking` requirements

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def get_extras_require() -> Dict[str, List[str]]:
     requirements = {
         # TODO(HideakiImamura) Unpin mypy version after fixing "Duplicate modules" error in
         # tutorials.
-        "checking": ["black", "hacking", "isort", "mypy==0.790", "blackdoc"],
+        "checking": ["black", "hacking", "isort", "mypy==0.790", "blackdoc", "toml"],
```
Comment (nzw0301) [inline setup.py:54]: The update in this PR might be removed in the near future, so could you add a TODO line like

```suggestion
        # TODO(xadrianzetx ) Remove `toml` when https://github.com/keewis/blackdoc/issues/100 is resolved
        "checking": ["black", "hacking", "isort", "mypy==0.790", "blackdoc", "toml"],
```
?

**Review by nzw0301 — COMMENTED** [review]
Awesome! Thank you for your investigation! According to https://github.com/keewis/blackdoc/issues/100, `blackdoc` will add `toml` a to its dependency list. So could you add a to-do line?

**Review by nzw0301 — APPROVED** [review]
👍🏻

**Comment by xadrianzetx** [comment]
@nzw0301 `blackdoc` [released a fix for that already](https://github.com/keewis/blackdoc/releases/tag/v0.3.4), so I'm going to close this one/

**Comment by nzw0301** [comment]
Thank you for the update!

**Pull request closed without merging**